src/Compute/Compute/Custom/VirtualMachinesOperationsExtensions.cs

```diff
@@ -13,12 +13,16 @@ public static partial class VirtualMachinesOperationsExtensions
         public static async Task<VirtualMachine> CreateOrUpdateWithCustomHeaderAsync(this IVirtualMachinesOperations operations, string resourceGroupName, string vmName, VirtualMachine virtualMachine, CancellationToken cancellationToken = default(CancellationToken))
         {
             var auxAuthHeader = virtualMachine.GetAuxAuthHeader();
+            var ifMatchResult = virtualMachine.GetIfMatchIfNoneMatch();
+            var ifMatch = ifMatchResult.Item1;
+            var ifNoneMatch = ifMatchResult.Item2;
```

Comment:
Is there a better way to retrieve the ifMatch and ifNoneMAtch? Maybe instead of making it a tuple, make it a class that has better named properties? OR can the name of the properties in the tuple be made to be more descriptive?

Reply:
i could retrieve them individually with seprate functions, but I just wanted to make it one function so I went with a Tuple. 
Item1, Item2 in this case are just getter methods for tuples, not named properties